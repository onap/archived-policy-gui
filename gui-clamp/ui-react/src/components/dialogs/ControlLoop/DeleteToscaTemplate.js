/*
 * ============LICENSE_START=======================================================
 * Copyright (C) 2021 Nordix Foundation.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * ============LICENSE_END=========================================================
 */

import React, { useState } from "react";
import Button from "react-bootstrap/Button";
import ControlLoopService from "../../../api/ControlLoopService";

const DeleteToscaTemplate = props => {

  const deleteTemplateHandler = async () => {
    console.log('deleteTemplateHandler called');

    const response = await ControlLoopService.deleteToscaTemplate(props.templateName, props.templateVersion)
      .catch(error => error.message);

    console.log('Response is ok: ' + response.ok);

    props.onDeleteToscaServiceTemplate(response);

  }

  return (
    <React.Fragment>
      <Button variant="danger"
              type="submit"
              style={{float: "right"}}
              onClick={ deleteTemplateHandler }>Delete Tosca Service Template</Button>
    </React.Fragment>
  );


}

export default DeleteToscaTemplate;
