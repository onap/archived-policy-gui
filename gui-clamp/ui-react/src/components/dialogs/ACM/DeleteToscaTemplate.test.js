/*
 * ============LICENSE_START=======================================================
 * Copyright (C) 2021 Nordix Foundation.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * ============LICENSE_END=========================================================
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import DeleteToscaTemplate from "./DeleteToscaTemplate";

describe('Verify DeleteToscaTemplate', () => {

  it("renders without crashing", () => {
    shallow(<DeleteToscaTemplate/>);
  });

  it("renders correctly", () => {
    const tree = shallow(<DeleteToscaTemplate/>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should have a Button element', () => {
    const container = shallow(<DeleteToscaTemplate/>)
    expect(container.find('Button').length).toEqual(1);
  });

  it('button should call deleteTemplateHandler when clicked', async () => {
    const component = mount(<DeleteToscaTemplate/>)
    const logSpy = jest.spyOn(console, 'log');

    component.find('[variant="danger"]').simulate('click');
    expect(logSpy).toHaveBeenCalledWith('deleteTemplateHandler called');
  });

  it('should have a Button element with specified text', () => {
    const container = shallow(<DeleteToscaTemplate/>)
    expect(container.find('Button').text()).toBe('Delete Tosca Service Template');
  });
});
