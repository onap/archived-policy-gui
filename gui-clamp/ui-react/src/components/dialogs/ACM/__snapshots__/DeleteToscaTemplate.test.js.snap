// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify DeleteToscaTemplate renders correctly 1`] = `
<Fragment>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "float": "right",
      }
    }
    type="submit"
    variant="danger"
  >
    Delete Tosca Service Template
  </Button>
</Fragment>
`;
